// Extractor service for Java <-> Native communication

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.tigerit.extractor.server.worker";
option java_outer_classname = "ExtractorServiceProto";

package extractor_service;

service ExtractorService {
	rpc Extract (InputFinger) returns (ExtractedFinger) {}
}

message InputFinger {
	int32 format = 1;
	bytes data = 2;
}

message ExtractedFinger {
	float quality = 1;
	bytes index = 2;
	bytes data = 3;
}
